readthedocs/projects/tasks.py

```diff
@@ -510,6 +512,26 @@ def run(
             self.commit = commit
             self.config = None
 
+            if self.project.has_feature(Feature.LIMIT_CONCURRENT_BUILDS):
+                response = api_v2.build.running.get(project__slug=self.project.slug)
+                if response.get('count', 0) >= settings.RTD_MAX_CONCURRENT_BUILDS:
+                    log.warning(
+                        'Delaying tasks due to concurrency limit. project=%s version=%s',
+                        self.project.slug,
+                        self.version.slug,
+                    )
+
+                    # This is done automatically on the environment context, but
+                    # we are executing this code before creating one
+                    api_v2.build(self.build['id']).patch({
+                        'error': BuildMaxConcurrencyError.message.format(
+                            limit=settings.RTD_MAX_CONCURRENT_BUILDS,
+                        ),
+                        'state': BUILD_STATE_QUEUED,
+                    })
+                    self.task.retry(exc=BuildMaxConcurrencyError, throw=False)
```

Comment:
Do we want to explicitly set a time here? We also likely want to set a large max number of retries.

Reply:
We can set this explicitly here. Currently, it's taking these numbers from the `@app.task` decorator at https://github.com/readthedocs/readthedocs.org/blob/759683771a2ccea2c439ee6327ac6754adf99cdc/readthedocs/projects/tasks.py#L403-L404

Reply:
What numbers do you think that work better for this case?

Reply:
Not sure, but we likely want almost infinite retries for concurrency I think? But that might have other issues if applied at the task level.

Reply:
I'm fine increasing the `max_retries` to something like 10 or 15. With 15 it will be Queued for almost 2hs (105m). So, the user will need to hit the concurrency for that time; which I think it's not common.

We are starting at 4, and it's per-project, not per-organization.

Reply:
Hrm... What I said is not 100% true, because the task will be re-queued and put it at the end of it. So, builders would need to be at full capacity (not taking this task) for that time.

Reply:
I updated the PR with `max_retries=25`